src/containers/Header/index.js

```diff
@@ -46,29 +66,104 @@ class Header extends Component {
       return;
     }
 
-    const desiredLink = ENV_LINK_MAP[mode] ?? process.env.REACT_APP_MAINNET_LINK;
+    const desiredLink = STATIC_ENV_LINKS[mode] ?? process.env.REACT_APP_MAINNET_LINK;
+    this.switchMode(desiredLink);
+  };
+
+  handleCustomNetworkClick = event => {
+    const currentRippledUrl = this.context; // this is undefined if not in sidechain mode
+    const newRippledUrl = event.currentTarget.getAttribute('value');
+
+    if (newRippledUrl.toLowerCase() === currentRippledUrl.toLowerCase()) {
+      return;
+    }
+
+    this.switchMode(`${SIDECHAIN_BASE_LINK}/${newRippledUrl}`);
+  };
+
+  onInputKeyDown = event => {
+    if (event.key === 'Enter') {
+      const currentRippledUrl = this.context;
+      const rippledUrl = event.currentTarget.value.trim();
+      if (
+        currentRippledUrl != null &&
+        rippledUrl.toLowerCase() === currentRippledUrl.toLowerCase()
+      ) {
+        return;
+      }
+      this.switchMode(`${SIDECHAIN_BASE_LINK}/${rippledUrl}`);
+    }
+  };
+
+  switchMode = desiredLink => {
     analytics(ANALYTIC_TYPES.event, {
       eventCategory: 'mode switch',
       eventAction: desiredLink,
     });
-    window.location = desiredLink;
+    window.location.assign(desiredLink);
   };
 
   ignore = event => {
     event.preventDefault();
     event.stopPropagation();
   };
 
+  renderDropdown = (network, handleClick, classname, text) => {
+    return (
+      <div
+        key={network}
+        className={classname}
+        role="menuitem"
+        value={network}
+        onClick={handleClick}
+        onKeyUp={handleClick}
+        tabIndex={0}
+      >
+        <span>{text}</span>
+        {/*
+        Adding this space to provide width between name
+        and selected checkmark
+        */}
+        &nbsp;
```

Comment:
Why not just add css padding between name and checkmark?

Reply:
@ledhed2222 added this

Reply:
seemed semantic? 🤷

Reply:
Haha ok. I'll approve but we should probably fix this in another PR.